src/searching/src/dijkstra.cairo

```diff
@@ -0,0 +1,202 @@
+//! Dijkstra algorithm using priority queue
+use core::traits::Index;
+use core::option::OptionTrait;
+use core::array::SpanTrait;
+use core::box::BoxTrait;
+use array::{Array, ArrayTrait};
+use traits::Into;
+use dict::Felt252DictTrait;
+use nullable::{NullableTrait, nullable_from_box, match_nullable, FromNullableResult};
+use alexandria_data_structures::vec::{Felt252Vec, VecTrait};
```

Comment:
where is that used ?

Reply:
Hello, thx a lot for the review.

- use traits::Into; is needed like in line 50.
- use dict::Felt252DictTrait; is used to call insert, squash, get on dictionnary. See Graph struct.
- I rework use nullable::{NullableTrait, nullable_from_box, match_nullable, FromNullableResult}; to only import use nullable::{FromNullableResult};
- use alexandria_data_structures::vec::{Felt252Vec, VecTrait}; is removed

Let me know if you're think is not ok.

Reply:
perfect